pkg/cmd/repo/sync/sync.go

```diff
@@ -0,0 +1,325 @@
+			a hard reset.
+
+			Without an argument, the local repository is selected as the destination repository.
+
+			By default the source repository is the parent of the destination repository,
+			this can be overridden with the %[1]s--source%[1]s flag.
+
+			The source repository default branch is selected automatically, but can be
+			overridden with the %[1]s--branch%[1]s flag.
+		`, "`"),
+		Example: heredoc.Doc(`
+			# Sync local repository from remote parent
+			$ gh repo sync
+
+			# Sync local repository from remote parent on non-default branch
+			$ gh repo sync --branch v1
+
+			# Sync remote fork from remote parent
+			$ gh repo sync owner/cli-fork
+
+			# Sync remote repo from another remote repo
+			$ gh repo sync owner/repo --source owner2/repo2
+		`),
+		Args: cobra.MaximumNArgs(1),
+		RunE: func(c *cobra.Command, args []string) error {
+			if len(args) > 0 {
+				opts.DestArg = args[0]
+			}
+			if runF != nil {
+				return runF(&opts)
+			}
+			return syncRun(&opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.SrcArg, "source", "s", "", "Source repository")
+	cmd.Flags().StringVarP(&opts.Branch, "branch", "b", "", "Branch to sync")
+	cmd.Flags().BoolVarP(&opts.Force, "force", "", false, "Discard destination repository changes")
+	return cmd
+}
+
+func syncRun(opts *SyncOptions) error {
+	if opts.DestArg == "" {
+		return syncLocalRepo(opts)
+	} else {
+		return syncRemoteRepo(opts)
+	}
+}
+
+func syncLocalRepo(opts *SyncOptions) error {
+	var err error
+	var srcRepo ghrepo.Interface
+
+	dirtyRepo, err := opts.Git.IsDirty()
+	if err != nil {
+		return err
+	}
+	if dirtyRepo {
+		return fmt.Errorf("can't sync because there are local changes, please commit or stash them")
+	}
+
+	if opts.SrcArg != "" {
+		srcRepo, err = ghrepo.FromFullName(opts.SrcArg)
+	} else {
+		srcRepo, err = opts.BaseRepo()
+	}
+	if err != nil {
+		return err
+	}
+
+	// Find remote that matches the srcRepo
+	var remote string
+	remotes, err := opts.Remotes()
+	if err != nil {
+		return err
+	}
+	for _, r := range remotes {
+		if r.RepoName() == srcRepo.RepoName() &&
+			r.RepoOwner() == srcRepo.RepoOwner() &&
+			r.RepoHost() == srcRepo.RepoHost() {
+			remote = r.Name
+		}
+	}
+	if remote == "" {
+		return fmt.Errorf("can't find corresponding remote for %s", ghrepo.FullName(srcRepo))
+	}
+
+	if opts.Branch == "" {
+		httpClient, err := opts.HttpClient()
+		if err != nil {
+			return err
+		}
+		apiClient := api.NewClientFromHTTP(httpClient)
+		opts.IO.StartProgressIndicator()
+		opts.Branch, err = api.RepoDefaultBranch(apiClient, srcRepo)
+		opts.IO.StopProgressIndicator()
+		if err != nil {
+			return err
+		}
+	}
+
+	opts.IO.StartProgressIndicator()
+	err = executeLocalRepoSync(srcRepo, remote, opts)
+	opts.IO.StopProgressIndicator()
+	if err != nil {
+		if errors.Is(err, divergingError) {
+			return fmt.Errorf("can't sync because there are diverging changes, you can use `--force` to overwrite the changes")
```

Comment:
Nit: split the sentences
```suggestion
			return fmt.Errorf("can't sync because there are diverging changes; use `--force` to overwrite the destination branch")
```